commit a343598dab021de428723cbb4ec45fc4fc3d80a3
Author: Jappie Klooster <jappieklooster@hotmail.com>
Date:   Sun Jun 25 15:52:08 2023 -0400

    Fix build for ghc 9.4.4
    
    These changes make it build once more

diff --git a/Setup.hs b/Setup.hs
index 729cdd7..e7d4aaf 100644
--- a/Setup.hs
+++ b/Setup.hs
@@ -169,7 +169,7 @@ main = do
 #else
           newHsc buildInfo localBuildInfo =
 #endif
-              PreProcessor {
+              ppUnlit {
                   platformIndependent = platformIndependent (origHsc buildInfo),
                   runPreProcessor = \inFiles outFiles verbosity -> do
                       llvmConfig <- getLLVMConfig (configFlags localBuildInfo)
diff --git a/src/LLVM/Internal/Operand.hs b/src/LLVM/Internal/Operand.hs
index 5de79bb..63866c8 100644
--- a/src/LLVM/Internal/Operand.hs
+++ b/src/LLVM/Internal/Operand.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE
   DuplicateRecordFields,
+  DisambiguateRecordFields,
   MultiParamTypeClasses,
   NamedFieldPuns,
   OverloadedStrings,
@@ -835,10 +836,18 @@ instance EncodeM EncodeAST A.DILocalVariable (Ptr FFI.DILocalVariable) where
     Context c <- gets encodeStateContext
     FFI.upCast <$> liftIO (FFI.getDILocalVariable c scope name file line type' arg flags alignInBits)
 
+getDITemplateParameterName :: A.DITemplateParameter -> ShortByteString
+getDITemplateParameterName A.DITemplateTypeParameter {..} = name
+getDITemplateParameterName A.DITemplateValueParameter{..} = name
+
+getDITemplateParameterType :: A.DITemplateParameter -> Maybe (A.MDRef A.DIType)
+getDITemplateParameterType A.DITemplateTypeParameter {..} = type'
+getDITemplateParameterType A.DITemplateValueParameter{..} = type'
+
 instance EncodeM EncodeAST A.DITemplateParameter (Ptr FFI.DITemplateParameter) where
   encodeM p = do
-    name' <- encodeM (A.name (p :: A.DITemplateParameter)) :: EncodeAST (Ptr FFI.MDString)
-    ty <- encodeM (A.type' (p :: A.DITemplateParameter))
+    name' <- encodeM (getDITemplateParameterName p) :: EncodeAST (Ptr FFI.MDString)
+    ty <- encodeM (getDITemplateParameterType p)
     Context c <- gets encodeStateContext
     case p of
       A.DITemplateTypeParameter {} ->
